dss/operations/events.py

```diff
@@ -86,24 +88,49 @@ def forward_keys(bundle_fqids):
 @events.action("journal",
                arguments={"--prefix": dict(required=True,
                                            help="flashflood prefix to journal events"),
-                          "--minimum-number-of-events": dict(default=1000, type=int),
-                          "--minimum-size": dict(default=1024 * 1024, type=int)})
+                          "--number-of-events": dict(default=None, type=int),
+                          "--starting-journal-id": dict(default=None),
+                          "--job-id": dict(default=None)})
 def journal(argv: typing.List[str], args: argparse.Namespace):
-    ff = FlashFlood(resources.s3, Config.get_flashflood_bucket(), args.prefix)
-    while True:
+    job_id = args.job_id or f"{uuid4()}"
+    cmd_template = (f"events journal --job-id {job_id} "
+                    f"--prefix {args.prefix} "
+                    f"--number-of-events {args.number_of_events} "
+                    f"--starting-journal-id {{}}")
+
+    if args.starting_journal_id is None:
+        start_time = datetime.now()
+        with SQSMessenger(command_queue_url) as sqsm:
+            journals = list()
+            for journal_id in list_new_flashflood_journals(args.prefix):
+                journals.append(journal_id)
+                if args.number_of_events == len(journals):
+                    print(f"Journaling from {journals[0]} with {args.number_of_events} events.")
+                    sqsm.send(cmd_template.format(journals[0]))
+                    journals = list()
+        monitor_logs(logs, job_id, start_time)
+    else:
+        msg = dict(action="journal", job_id=job_id, prefix=args.prefix, key=args.starting_journal_id)
         try:
-            ff.journal(args.minimum_number_of_events, args.minimum_size)
-        except FlashFloodJournalingError:
-            break
+            journal_flashflood(args.prefix, args.number_of_events, JournalID(args.starting_journal_id))
```

Comment:
The `journal_flashflood()` method compiles a list of Journals, then assembles and returns a combined journal. But that value isn't assigned to a variable here, so it doesn't seem like this `journal_flashflood()` call actually does anything. Can you clarify what's happening with this function call?

Reply:
The heavy lifting is performed by `ff.combine_journals()`, which causes flashflood to save the new journal to s3.